machineconfiguration/v1/types.go

```diff
@@ -575,6 +642,14 @@ const (
 	// MachineConfigPoolRenderDegraded means the rendered configuration for the pool cannot be generated because of an error
 	MachineConfigPoolRenderDegraded MachineConfigPoolConditionType = "RenderDegraded"
 
+	// MachineConfigPoolPinnedImageSetsDegraded means the pinned image sets for the pool cannot be populated because of an error
+	// +openshift:enable:FeatureGate=PinnedImages
+	MachineConfigPoolPinnedImageSetsDegraded MachineConfigPoolConditionType = "PinnedImageSetsDegraded"
+
+	// MachineConfigPoolSynchronizerDegraded means the pool synchronizer can not be updated because of an error
+	// +openshift:enable:FeatureGate=PinnedImages
+	MachineConfigPoolSynchronizerDegraded MachineConfigPoolConditionType = "PoolSynchronizerDegraded"
```

Comment:
Just to understand this degrade, for example, for pinnedimagesets, how would this differ from `MachineConfigPoolPinnedImageSetsDegraded`?

For example, is `MachineConfigPoolPinnedImageSetsDegraded` -> a pull failed because not enough space on disk,
and `MachineConfigPoolSynchronizerDegraded` -> there's unavailable machines for that type (so more generic)?

Reply:
well in this context the degraded types would be for the controllers that populate the data for MCP. So for example the pinnedImageSet controller would just populate the MCP spec. If the controller was degraded we would say why. THis is the same pattern used by for example the render controller.

So I would expect a small controller that watches on the synchronizer type to populate the status. If more types are created that controller would watch another resource. 

So these are status for controllers maintaining MCP vs status for the synchronization actions if that makes sense.